### ISSUE TITLE
Easy test token minting in the web ui

### ISSUE THREAD

**Issue by hackaugusto** [raiden-network/raiden#2471]
Expose the mint button if the token supports it, this should make funding accounts for testing easier.

**Comment by hackaugusto** [comment]
IMO this should be expose regardless of the chain being used.

**Comment by LefterisJP** [comment]
> IMO this should be expose regardless of the chain being used.

Why? We can conditionally expose it only on testnet. On mainnet this makes no sense.

**Comment by LefterisJP** [comment]
> If someone deployed a token with a mint function

So you want to make an extra RPC call for each token contract in the mainnet to see if there is a function matching this ABI: `void mint()` and then expose this to the webui if some token contract does that? This is the only way I know to check if a contract has a function. Try and invoke it (in a `call()` not in a `transact()`, so that no transaction gets sent).

But mint is not part of the ERC20 standard. And nobody tests stuff like that in the mainnet. It's expensive.  Plus if we do that the raiden client will be making an assumption. That interacting with a contract's `mint()` function is safe. Someone could write a function with the same interface() that doesn't mint tokens but does something completely different.

The only way I can imagine this working is if we have pre-approved contracts we know have a `mint()` function in the mainnet. We could do that. But at the moment I know of no such contract.

**Comment by hackaugusto** [comment]
> So you want to make an extra RPC call for each token contract in the mainnet to see if there is a function matching this ABI: void mint() and then expose this to the webui if some token contract does that? This is the only way I know to check if a contract has a function. Try and invoke it (in a call() not in a transact(), so that no transaction gets sent).

Yes, this is how I imagine this works regardless of the network.
 
>  That interacting with a contract's mint() function is safe. Someone could write a function with the same interface() that doesn't mint tokens but does something completely different.

That is true for any ERC20 complaint contract too.

> The only way I can imagine this working is if we have pre-approved contracts we know have a mint() function in the mainnet. We could do that. But at the moment I know of no such contract.

https://github.com/raiden-network/raiden/issues/344

**Comment by LefterisJP** [comment]
> Yes, this is how I imagine this works regardless of the network.

I just argue that this is not required in the mainnet and not worth the extra RPC call. Virtually no token I know of has this function.

I would instead suggest use a whitelist of contracts that we know of have mint() and expose the mint() call in the webui only for such contracts. This whitelist should be a configuration option. This way it would be easy to change between testnet and mainnet.

#344 may be an overkill for the way to implement this whitelist.
